### PR TITLE
New -XX[+|-]ShareAnonymousClasses command line option

### PR DESCRIPTION
**Pull request by pmhayward**
Signed-off-by: Peter Hayward <pmhayward@uk.ibm.com>

### CODE REVIEW

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

File: docs/version0.16.md [hunk 1]
```diff
@@ -29,26 +29,42 @@
 
 - [New binaries and changes to supported environments](#binaries-and-supported-environments)
 - [Automatic setting of initial heap size is enabled by default](#automatic-setting-of-initial-heap-size-is-enabled-by-default)
-
-
+- [Option to share anonymous classes](#option-to-share-anonymous-classes)
```
Comment (DanHeidinga) [inline docs/version0.16.md]:
```suggestion
- [Option to share VM anonymous classes](#option-to-share-anonymous-classes)
```

File: docs/version0.16.md [hunk 2]
```diff
@@ -29,26 +29,42 @@
...
+To learn more about support for OpenJ9 releases, including OpenJDK levels and platform support, see [Supported environments](openj9_support.md).
 
 ### Automatic setting of initial heap size is enabled by default
 
- OpenJ9 version 0.15.1 introduced the [`-XX:[+|-]UseGCStartupHints`](xxusegcstartuphints.md) option, which, when enabled, turned on the automatic learning and setting of an appropriate heap size for an application. This option is now enabled by default.
+OpenJ9 version 0.15.1 introduced the [`-XX:[+|-]UseGCStartupHints`](xxusegcstartuphints.md) option, which, when enabled, turned on the automatic learning and setting of an appropriate heap size for an application. This option is now enabled by default.
+
+### Option to share anonymous classes
+
+Prior to version 0.16.0, anonymous classes were not stored in the shared class cache. They are now stored there by default, which means they are available for ahead-of-time (AOT) compilation, potentially improving startup performance. A new command, [-XX:[+|-]ShareAnonymousClasses](xxshareanonymousclasses.md), is introduced that enables you to stop anonymous classes being stored in the shared class cache.
```
Comment (DanHeidinga) [inline docs/version0.16.md]:
```suggestion
Prior to version 0.16.0, anonymous classes, those created by Unsafe.defineAnonymousClass, were not stored in the shared class cache. They are now stored there by default, which means they are available for ahead-of-time (AOT) compilation, potentially improving startup performance. A new command, [-XX:[+|-]ShareAnonymousClasses](xxshareanonymousclasses.md), is introduced that enables you to stop anonymous classes being stored in the shared class cache.

The current implementation only shares anonymous classes that are created for Java 8 lambda expressions.  This limitation will be removed in a future release.
```

File: docs/xxshareanonymousclasses.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:[+|-]ShareAnonymousClasses
+
+This option enables and disables the storage of anonymous classes in the shared class cache.
```
Comment (DanHeidinga) [inline docs/xxshareanonymousclasses.md]:
```suggestion
This option enables and disables the storage of VM anonymous classes in the shared class cache.
```

File: docs/xxshareanonymousclasses.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+The option is enabled by default, which means that methods in anonymous classes are stored in the shared class cache and are therefore available for ahead-of-time (AOT) compilation, potentially improving startup performance.
```
Comment (DanHeidinga) [inline docs/xxshareanonymousclasses.md]:
```suggestion
The option is enabled by default, which means anonymous classes are stored in the shared class cache and are therefore available for ahead-of-time (AOT) compilation, potentially improving startup performance.
```

**Comment by DanHeidinga** [comment]
PR for #336

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by DanHeidinga** [comment]
@pmhayward I see you resolved the review comments but didn't address the missing "VM" and Unsafe.defineAnonymousClass context.  Was that intentional?

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
@DanHeidinga : Apologies! Omitting changes wasn't intentional; just my incompetence with Git commands :) If I've got it right now, should be correctly restaged at:
https://eclipse.github.io/openj9-docs-staging/337/version0.16/#option-to-share-vm-anonymous-classes

**Comment by DanHeidinga** [comment]
No worries @pmhayward.  I figured it was unintentional.

Current changes look good to me apart from the copyright date nitpick

File: docs/xxshareanonymousclasses.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<!--
+* Copyright (c) 2017, 2019 IBM Corp. and others
```
[the remark this answers is not in the document]
Comment (pmhayward) [reply docs/xxshareanonymousclasses.md]: The docs have always been treated as a "collection" and considered a single entity as far as copyright is concerned. The copyright dates are therefore the same across all files, regardless of when an individual file was created.

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
Jenkins doc stage

**Pull request merged**